doc/UltiSnips.txt

```diff
@@ -954,6 +964,49 @@ The 'snip' object provides some properties as well: >
         'start' - placeholder start on the moment of selection;
         'end' - placeholder end on the moment of selection;
 
+
+The 'snip.store' object provides dict-like stores to remember variables accross
+snippet executions: >
+
+    snip.store.tmp:
+        A temporary local store accessible only during the snippet execution
+        and updates in the same python code section. It is reset each time the snippet is called, but not
+        between updates.
+
+    snip.store.snippet:
+        A temporary local store accessible only during the snippet execution
+        and updates in all python code sections. It is reset each time the snippet is called, but not
+        between updates.
+
+    snip.store.buffer:
+        A store accessible to all snippets executed in the same buffer. Each
+        buffer has its own one.
+
+    snip.store.session:
+        A store accessible to all snippets in a same vim instance.
+
+    snip.store.file:
+        A persistent store attached to a file. Any snippet executed on the
+        same file (possibly on different vim instances) will access the same
+        store. If g:UltiSnipStoreDir is unset or set to v:null, then it behave
+        similary to store.buffer.
+
+    snip.store.common:
+        A persistent global store common to all snippet executions in all buffers and any
+        vim instance. If g:UltiSnipStoreDir is unset or set to v:null, then it behave
+        similary to store.session.
+
+The stores have some syntax sugars compared to built-in dicts : >
+    store['key', default]
```

Comment:
I find the syntax for default unusual and unexpected. I'd suggest to instead use `get` to keep symmetry with Python's dictionary. The downside is, that the access to the store now needs 4 strokes more. Maybe we can come up with a shorter form instead of `snip.store.<type>`